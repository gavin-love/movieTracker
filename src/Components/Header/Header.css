.header {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 150px;
  background-color: #87aaae;
  box-shadow: 0px 2px 10px 0px rgba(0, 0, 0, 0.3);
}

a {
  color: white;
  margin-right: 2%;
  text-decoration: none;
  font-weight: 200;
}
a:hover {
  color: #bc4676
}

.active {
   color: #bc4676
}

.nav_links_container {
  display: flex;
  justify-content: flex-end;
  height: 20px;
  padding-bottom: 1%;
}

.logo_section {
  height: 130px;
  background-image: url("../../Assets/Movie-Tracker.png");
  background-repeat: no-repeat;
  background-size: 20%;
  background-position: center;
}
