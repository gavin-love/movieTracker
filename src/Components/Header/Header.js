import React from "react";
import { NavLink } from "react-router-dom";
import { connect } from "react-redux";
import { addLogout } from "../../Actions/index";
import "./Header.css";

export const Header = props => {
  const signoutButton = (
    <NavLink to="/" onClick={props.handleLogout}>
      Sign-Out
    </NavLink>
  );
  return (
    <div className="header">
      <NavLink to="/">Home</NavLink>
      <NavLink to="/favorites">Favorites</NavLink>
      {props.user.user_id ? (
        signoutButton
      ) : (
        <NavLink to="/login">Sign-In</NavLink>
      )}
    </div>
  );
};

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  handleLogout: () => dispatch(addLogout())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
