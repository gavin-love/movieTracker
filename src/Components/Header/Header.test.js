import React from 'react';
import { Header } from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  it('should match the snapshot', () => {
    let mockProps = {
      user: {
        user_id: 1
      }
    }
    const wrapper = shallow(<Header {...mockProps} />)
    expect(wrapper).toMatchSnapshot();
  })
})