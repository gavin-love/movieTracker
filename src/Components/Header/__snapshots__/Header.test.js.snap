// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    user={
        Object {
            "user_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="logo_section"
/>,
        <div
          className="nav_links_container"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/"
          >
                    Home
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    to="/favorites"
          >
                    Favorites
          </NavLink>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    onClick={[Function]}
                    to="/"
          >
                    Sign-Out
          </NavLink>
</div>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "logo_section",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/"
>
              Home
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              to="/favorites"
>
              Favorites
</NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              onClick={[Function]}
              to="/"
>
              Sign-Out
</NavLink>,
          ],
          "className": "nav_links_container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Favorites",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Sign-Out",
              "onClick": [Function],
              "to": "/",
            },
            "ref": null,
            "rendered": "Sign-Out",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="logo_section"
/>,
          <div
            className="nav_links_container"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/"
            >
                        Home
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        to="/favorites"
            >
                        Favorites
            </NavLink>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        onClick={[Function]}
                        to="/"
            >
                        Sign-Out
            </NavLink>
</div>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "logo_section",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/"
>
                Home
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                to="/favorites"
>
                Favorites
</NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                onClick={[Function]}
                to="/"
>
                Sign-Out
</NavLink>,
            ],
            "className": "nav_links_container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Favorites",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Sign-Out",
                "onClick": [Function],
                "to": "/",
              },
              "ref": null,
              "rendered": "Sign-Out",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
